### ISSUE TITLE
ViroSample Navigate From Child ViroARScene to Parent ViroARSceneNavigator

### ISSUE THREAD

**Issue by WillGeller** [viromedia/viro#174]
I would like to know how to modify the ViroSample to allow you jump back to the AR/VR scene selector from a button on the AR Scene. I tried using parent child communication but I realize the child component defined in `HelloWorldSceneAR.js` doesn't seem to be able to access props declared on the navigator. I think the way the example is set up makes it hard to pass data from the child component scene up to the navigator parent component.

**Comment by WillGeller** [comment]
Full Code example:
App.js
```
/**
 * Copyright (c) 2017-present, Viro, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

import React, { Component } from 'react';
import {
  AppRegistry,
  Text,
  View,
  StyleSheet,
  PixelRatio,
  TouchableHighlight,
} from 'react-native';

import {
  ViroSceneNavigator,
  ViroARSceneNavigator
} from 'react-viro';

var sharedProps = {
  apiKey:"",
  debug: true
}

// Sets the default scene you want for AR and VR
var portalScene1 = require('./js/PortalScene1')
var portalScene2 = require('./js/PortalScene2');

var PICK = "PICK";
var PORTAL_SCENE_1 = "PORTAL_SCENE_1";
var PORTAL_SCENE_2 = "PORTAL_SCENE_2";

// This determines which type of experience to launch in
var defaultNavigatorType = PICK;

export default class ViroSample extends Component {
  constructor() {
    super();

    this.state = {
      navigatorType : defaultNavigatorType,
      sharedProps : sharedProps
    }
    
    this._getExperienceSelector = this._getExperienceSelector.bind(this);
    this._getARPortal = this._getARPortal.bind(this);
    this._getExperienceButtonOnPress = this._getExperienceButtonOnPress.bind(this);
    this._exitScene = this._exitScene.bind(this);
  }

  render() {
    if (this.state.navigatorType == PICK) {
      return this._getExperienceSelector();
    } else if (this.state.navigatorType == PORTAL_SCENE_1) {
      return this._getARPortal(portalScene1);
    } else if (this.state.navigatorType == PORTAL_SCENE_2) {
      return this._getARPortal(portalScene2);
    }
  }

  // Presents the user with a choice of a Portal Scene
  _getExperienceSelector() {
    return (
      <View style={localStyles.outer} >
        <View style={localStyles.inner} >

          <Text style={localStyles.titleText}>
            Choose your desired experience:
          </Text>

          <TouchableHighlight style={localStyles.buttons}
            onPress={this._getExperienceButtonOnPress(PORTAL_SCENE_1)}
            underlayColor={'#68a0ff'} >

            <Text style={localStyles.buttonText}>AR1</Text>
          </TouchableHighlight>

          <TouchableHighlight style={localStyles.buttons}
            onPress={this._getExperienceButtonOnPress(PORTAL_SCENE_2)}
            underlayColor={'#68a0ff'} >

            <Text style={localStyles.buttonText}>AR2</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }

  // Returns the ViroARSceneNavigator which will start the AR experience
  _getARPortal(scene) {
    return (
      <ViroARSceneNavigator {...this.state.sharedProps}
        initialScene={{scene: scene}}
        onExit={this._exitScene} />
    );
  }

  // This function returns an anonymous/lambda function to be used
  // by the experience selector buttons
  _getExperienceButtonOnPress(navigatorType) {
    return () => {
      this.setState({
        navigatorType : navigatorType
      })
    }
  }

  // This function "exits" AR Portal Scene by setting the navigatorType to PICK.
  _exitScene() {
    this.setState({
      navigatorType : PICK
    })
  }
}

var localStyles = StyleSheet.create({
  viroContainer :{
    flex : 1,
    backgroundColor: "black",
  },
  outer : {
    flex : 1,
    flexDirection: 'row',
    alignItems:'center',
    backgroundColor: "black",
  },
  inner: {
    flex : 1,
    flexDirection: 'column',
    alignItems:'center',
    backgroundColor: "black",
  },
  titleText: {
    paddingTop: 30,
    paddingBottom: 20,
    color:'#fff',
    textAlign:'center',
    fontSize : 25
  },
  buttonText: {
    color:'#fff',
    textAlign:'center',
    fontSize : 20
  },
  buttons : {
    height: 80,
    width: 150,
    paddingTop:20,
    paddingBottom:20,
    marginTop: 10,
    marginBottom: 10,
    backgroundColor:'#68a0cf',
    borderRadius: 10,
    borderWidth: 1,
    borderColor: '#fff',
  },
  exitButton : {
    height: 50,
    width: 100,
    paddingTop:10,
    paddingBottom:10,
    marginTop: 10,
    marginBottom: 10,
    backgroundColor:'#68a0cf',
    borderRadius: 10,
    borderWidth: 1,
    borderColor: '#fff',
  }
});

module.exports = ViroSample
```

PortalScene1.js
```
'use strict';

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {StyleSheet} from 'react-native';

import {
  ViroSceneNavigator,
  ViroScene,
  ViroARScene,
  ViroAmbientLight,
  Viro360Video,
  Viro360Image,
  ViroUtils,
  ViroPortal,
  ViroPortalScene,
  Viro3DObject,
  ViroText
} from 'react-viro';

var createReactClass = require('create-react-class');
var MainScene = createReactClass({

  propTypes: {
      onExit:   PropTypes.func
  },

  render: function() {
    return (
      <ViroARScene>
      <ViroAmbientLight color="#ffffff" intensity={300}/>
        <ViroPortalScene passable={true}>
          <ViroPortal position={[0, 0, -1]} scale={[.1, .1, .1]}>
            <Viro3DObject source={require('./src/portal_ship/portal_ship.vrx')}
              resources={[require('./src/portal_ship/portal_ship_diffuse.png'),
                          require('./src/portal_ship/portal_ship_normal.png'),
                          require('./src/portal_ship/portal_ship_specular.png')]}
              type="VRX"/>
          </ViroPortal>
          <Viro360Image source={require("./src/360_island.jpg")} rotation={[0, 75, 0]}/>
        </ViroPortalScene>

        <ViroText
            text="BACK" width={2} height={2} position={[0, -3, -4]} style={styles.helloWorldTextStyle} 
            onClick={this._goBack}
        />
      </ViroARScene>
    );
  },

  _goBack() {
    console.log(this.props.onExit)
    if (typeof this.props.onExit === 'function') {
      this.props.onExit(e.target.value);
    }
  }
});

var styles = StyleSheet.create({
  helloWorldTextStyle: {
    fontFamily: 'Arial',
    fontSize: 25,
    color: '#ffffff',
    textAlignVertical: 'center',
    textAlign: 'center',  
  },
});

module.exports = MainScene;
```

**Comment by manbod** [comment]
Hey @WillGeller,
Thanks for sending complete code to help debug. We provide some basic scene navigation operations for switching between scenes on our `ViroSceneNavigator` and similarly on `ViroARSceneNavigator` objects. You can refer [here](https://docs.viromedia.com/docs/scene-navigation) for future reference.

In this case though, if I understand what you are trying to achieve correctly, you could modify your `_getARPortal(scene)` something on the lines as follows:
```javascript
  _getARPortal(scene) {
    return (
      <View style={localStyles.viroContainer} >
        <ViroARSceneNavigator
          {...this.state.sharedProps}
          initialScene={{scene: scene}} />
         
        <View style={{position: 'absolute', bottom: 20, alignItems: 'center'}}>
          <TouchableHighlight style={localStyles.exitButton}
            onPress={()=>{this._exitScene}}
            underlayColor={'#00000000'} >
            <Text style={localStyles.buttonText}>Exit</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
``` 
So basically, you can have a screen in your app where ViroReact and simple React Native views reside along side each other and interact based on app states. In the above snippet, you'll render a React Native View with two children 1) ViroARSceneNavigator and 2) An exit button.

Hope this helps! Let us know if you have more questions.

Thanks!

P.S.: As an example, you can also checkout an example app we launched that has similar UI concepts (both ViroReact and React Native views) - [Figment AR](https://itunes.apple.com/us/app/figment-ar/id1270018902?mt=8)

**Comment by WillGeller** [comment]
Thank you! That does accomplish navigation using React UI elements outside of the Viro View. I will look into using Redux or something to pass event Data out of the Scene View.
